src/sql/base/parts/editableDropdown/browser/dropdown.ts

```diff
@@ -209,8 +214,11 @@ export class Dropdown extends Disposable {
 		});
 
 		this._controller.onDropdownEscape(() => {
```

Comment:
Is this also triggered when the dropdown is closed via other means such as selecting a value?

Reply:
No, it is only triggered when pressing the escape button.

Reply:
That doesn't seem right then to me - shouldn't the behavior be the same?

Reply:
I guess, we can add the same code to onSelectionChange()

Reply:
the requirement of restoring keyboard focus is only required for escape key, when a value is selected, the focus should go to the textbox

Reply:
When you say "the requirement" what are you referring to? Because I don't think we should take that issue as being the full description of correct behavior, it was calling out behavior that was specifically wrong but I wouldn't necessarily assume they'd list all the different scenarios and their expected results. And having it behave differently when selecting an item vs escaping out of the dropdown seems odd to me.

Reply:
Where do you get the requirement of "when a value is selected, the focus should go to the textbox"?

Reply:
the "requirement" of restoring keyboard focus is for escape key press scenario, there isn't a requirement of restoring keyboard focus when selection done, so probably I should say it makes more sense for the focus to go to the input box. I took a look at the SSMS one, the arrow is not a focusable element, I think that is probably the desired behavior, when you click on it, it will bring up the list, and when you use keyboard, you can already bring up the list by using ALT+Down key, what do you guys think? @Charles-Gagnon  @aasimkhan30

Reply:
In the input box you're saying? That seems fine as long as we can make it clear through the screen reader that this is an expandable element.

Reply:
Are you referring to something like this? https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-none.html

Reply:
yes, this one, or the .NET editable combobox, they both are not exposing the dropdown arrow as a focusable element, but since we are using actionbar here, I am not sure how easy it is to make it not keyboard focusable.

Reply:
You should be able to just override it with styles to set the tab index on the anchor element.